File: backend/schemas/session.py
from pydantic import BaseModel, ConfigDict
from typing import Optional, Dict, List, Any

class SessionConfigSchema(BaseModel):
    max_rounds: int
    convergence_majority: float
    priority_weights: Dict[str, float]
    thought_inspector_enabled: bool

class SessionSchema(BaseModel):
    id: str
    topic: str
    supporting_context: Optional[str] = None
    status: str
    config: SessionConfigSchema
    created_at: str
    ended_at: Optional[str] = None
    termination_reason: Optional[str] = None
    rounds_elapsed: Optional[int] = None
    agent_count: Optional[int] = None

class SessionTemplateSchema(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    name: str
    description: Optional[str] = None
    agents: List[Dict[str, Any]]
    config: SessionConfigSchema
    created_at: str

File: shared/types/session.ts
export type SessionStatus = 'configured' | 'running' | 'paused' | 'ended';

export type TerminationReason = 'consensus' | 'cap' | 'host' | 'error' | null;

export interface SessionConfig {
  max_rounds: number;
  convergence_majority: number;
  priority_weights: {
    recency: number;
    novelty: number;
    role: number;
  };
  thought_inspector_enabled: boolean;
}

export interface Session {
  id: string;
  topic: string;
  supporting_context?: string;
  status: SessionStatus;
  config: SessionConfig;
  created_at: string;
  ended_at: string | null;
  termination_reason: TerminationReason;
  // Included in some endpoint responses like GET /sessions/{id}
  rounds_elapsed?: number;
  agent_count?: number;
}
